custom/sensors/SparkEncoder.java

```diff
@@ -95,40 +206,58 @@ public void setReverseDirection(boolean reverseDirection) {
 
     @Override
     public void reset() {
-        if (pidSource == PIDSourceType.kDisplacement) {
-            truePosition = encoder.getPosition();
-        } else {
-            trueVelocity = encoder.getVelocity();
-        }
+        truePosition = encoder.getPosition();
+        trueVelocity = encoder.getVelocity();
     }
 
     @Override
-    public double pidGetSafely() {
-        return pidGet();
+    public double pidGetSafely() throws InvalidSensorException {
+        switch (pidSource) {
+            case kDisplacement:
+                return getDistance();
+            case kRate:
+                return getRate();
+            default:
+                return getDistance();
+        }
     }
 
     @Override
-    public int getSafely() {
-        return get();
+    public int getSafely() throws InvalidSensorException {
```

Comment:
Can you add comment saying this is here because only of the bad super class requirement?